staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -225,40 +225,25 @@ func (h *DefaultPluginHandler) Execute(executablePath string, cmdArgs, environme
 // HandlePluginCommand receives a pluginHandler and command-line arguments and attempts to find
 // a plugin executable on the PATH that satisfies the given arguments.
 func HandlePluginCommand(pluginHandler PluginHandler, cmdArgs []string) error {
-	var remainingArgs []string // all "non-flag" arguments
-	for _, arg := range cmdArgs {
+	var pluginPath string
+	var pluginArgs []string
+	for i, arg := range cmdArgs {
 		if strings.HasPrefix(arg, "-") {
-			break
+			continue // flag
 		}
-		remainingArgs = append(remainingArgs, strings.Replace(arg, "-", "_", -1))
-	}
-
-	if len(remainingArgs) == 0 {
-		// the length of cmdArgs is at least 1
-		return fmt.Errorf("flags cannot be placed before plugin name: %s", cmdArgs[0])
-	}
-
-	foundBinaryPath := ""
-
-	// attempt to find binary, starting at longest possible name with given cmdArgs
-	for len(remainingArgs) > 0 {
-		path, found := pluginHandler.Lookup(strings.Join(remainingArgs, "-"))
-		if !found {
-			remainingArgs = remainingArgs[:len(remainingArgs)-1]
-			continue
+		for j, part := range cmdArgs[i:] {
+			if strings.HasPrefix(part, "-") {
+				break // flag
+			}
+			possiblePluginName := strings.Join(cmdArgs[i:i+1+j], "-")
```

Comment:
User has two plugins installed `kubectl-foo` and `kubectl-bar`;
Does this support this: `kubectl --namespace foo bar` when user wants to run `bar` plugin.

Although it seems an extreme case, in terms of namespaces maybe such a conflict is not extreme.

But I'd like to note that cobra already supports `kubectl --namespace set create` somehow.